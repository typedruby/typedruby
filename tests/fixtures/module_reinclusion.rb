# @typedruby

class Foo < Array
  include Enumerable
end

class Bar
  include Enumerable
  include Enumerable
end

module MonomorphicModule
end

class Bar
  include MonomorphicModule
  include MonomorphicModule
end
