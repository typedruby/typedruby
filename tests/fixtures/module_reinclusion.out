error: Duplicate include

        @ __ROOT__/tests/fixtures/module_reinclusion.rb:4
      4 |    include Enumerable::[Integer]
                     ^^^^^^^^^^^^^^^^^^^^^ here
        @ (builtin stdlib):__LINE__
    784 |    include Enumerable::[ElementType]
                     ^^^^^^^^^^^^^^^^^^^^^^^^^ previous inclusion was here

error: Duplicate include

        @ __ROOT__/tests/fixtures/module_reinclusion.rb:9
      9 |    include Enumerable::[String]
                     ^^^^^^^^^^^^^^^^^^^^ here
        @ __ROOT__/tests/fixtures/module_reinclusion.rb:8
      8 |    include Enumerable::[Integer]
                     ^^^^^^^^^^^^^^^^^^^^^ previous inclusion was here
